app/uk/gov/hmrc/exports/scheduler/jobs/PurgeAncientSubmissionsJob.scala

```diff
@@ -0,0 +1,129 @@
+  val appConfig: AppConfig,
+  submissionRepository: SubmissionRepository,
+  declarationRepository: DeclarationRepository,
+  unparsedNotificationRepository: UnparsedNotificationWorkItemRepository,
+  parsedNotificationRepository: ParsedNotificationRepository,
+  transactionalOps: PurgeSubmissionsTransactionalOps
+)(implicit ec: ExecutionContext)
+    extends ScheduledJob with ExportsClient with Logging with DecorateAsScala {
+
+  lazy val submissionCollection = db.getCollection(submissionRepository.collectionName)
+  lazy val declarationCollection = db.getCollection(declarationRepository.collectionName)
+  lazy val notificationCollection = db.getCollection(parsedNotificationRepository.collectionName)
+  lazy val unparsedNotificationCollection = db.getCollection(unparsedNotificationRepository.collectionName)
+
+  override val name: String = "PurgeAncientSubmissions"
+
+  override def interval: FiniteDuration = jobConfig.interval
+  override def firstRunTime: Option[LocalTime] = Some(jobConfig.elapseTime)
+
+  private def jsonSettings(mode: JsonMode) = JsonWriterSettings.builder.outputMode(mode).build
+
+  private val jobConfig = appConfig.purgeAncientSubmissions
+  private val clock: Clock = appConfig.clock
+
+  private val latestStatus = "latestEnhancedStatus"
+  private val statusLastUpdated = "enhancedStatusLastUpdated"
+
+  private val expiryDate = Codecs.toBson(ZonedDateTime.now(clock).minusDays(180))
+
+  private val latestStatusLookup =
+    in(latestStatus, List("GOODS_HAVE_EXITED", "DECLARATION_HANDLED_EXTERNALLY", "CANCELLED"): _*)
+
+  private val olderThanDate = lte(statusLastUpdated, expiryDate)
+
+  override def execute(): Future[Unit] = {
+
+    implicit val formatNotification: Format[ParsedNotification] = ParsedNotification.format
+    import ExportsDeclaration.Mongo._
+
+    val submissions: List[Submission] = submissionCollection
+      .find(and(olderThanDate, latestStatusLookup))
+      .asScala
+      .flatMap {
+        parseJsonFromDocument[Submission]
+      }
+      .toList
+
+    logger.info(s"Submissions found older than 180 days: ${submissions.size}")
+
+    val declarations: List[ExportsDeclaration] = declarationCollection
+      .find(in("id", submissions.map(_.uuid): _*))
+      .asScala
+      .flatMap {
+        parseJsonFromDocument[ExportsDeclaration]
+      }
+      .toList
+
+    logger.info(s"Declarations found linked to submissions: ${declarations.size}")
+
+    val parsedNotifications: List[ParsedNotification] = notificationCollection
+      .find(in("actionId", submissions.flatMap(_.actions.map(_.id)): _*))
+      .asScala
+      .flatMap {
+        parseJsonFromDocument[ParsedNotification]
+      }
+      .toList
+
+    logger.info(s"Parsed notifications found linked to submissions: ${parsedNotifications.size}")
+
+    val unparsedNotifications: List[UnparsedNotification] =
+      unparsedNotificationCollection
+        .find(in("item.id", parsedNotifications.map(_.unparsedNotificationId.toString): _*))
```

Comment:
Why do you need to retrieve the notifications? Once you have the Submissions you can directly remove the documents